### PR TITLE
Pull request to staging from dev

### PR DESCRIPTION
**Pull request by Phaedra144**

### CODE REVIEW

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/MergedCalController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+@RestController
+public class MergedCalController {
+
+  @Autowired
+  MergedCalendarRepository mergedCalendarRepository;
+
+  @PostMapping(value = "/calendar")
+  public ResponseEntity postMergedCal(@RequestBody MergedCalendar mergedCalendar) {
+    mergedCalendarRepository.save(mergedCalendar);
+    return new ResponseEntity(HttpStatus.OK);
```
Comment (Phaedra144) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/MergedCalController.java]: What if something goes wrong? ANyway we send httpStatusOk???

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/entity/UserModel.java [hunk 2]
```diff
@@ -56,4 +56,4 @@ public void setUserEmail(String userEmail) {
         this.userEmail = userEmail;
     }
 
-}
+}
```
Comment (Phaedra144) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/entity/UserModel.java]: Extra line is missing from here

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/AuthContoller.java [hunk 3]
```diff
@@ -29,12 +29,10 @@ public AuthResponse postAuth(@RequestBody AuthModel authModel, @RequestHeader("X
...
-            authAndUserService.saveUserModel(userModel);
         }
         authModel.setUser(userModel);
         authAndUserService.saveAuthModel(authModel);
-        AuthResponse authResponse = new AuthResponse(userModel.getId(), userModel.getClientToken(), authModel.getEmail(), authModel.getAccessToken());
 
-        return authResponse;
+        return new AuthResponse(userModel.getId(), userModel.getClientToken(), authModel.getEmail(), authModel.getAccessToken());
     }
 }
```
Comment (Phaedra144) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/AuthContoller.java]: Now I got "Column 'user_id' cannot be null" error, we have to think it over again.

Comment (Phaedra144) [reply src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/AuthContoller.java]: It is ok now!

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/MergedCalController.java (hunk 1 above)
Comment (Phaedra144) [reply src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/MergedCalController.java]: name of the method should be "postCalendar" based on our naming convention

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/entity/MergedCalendar.java [hunk 4]
```diff
@@ -11,38 +15,43 @@
 public class MergedCalendar {
 
     @Id
-    private long userId;
-    private String userName;
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
+    private UserModel user;
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/entity/MergedCalendar.java]: @gumimaci88 `@ManyToOne`

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/entity/UserModel.java [hunk 5]
```diff
@@ -13,11 +13,19 @@
     private String clientToken;
     private String userEmail;
 
+    List<MergedCalendar> mergedCalendarList;
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/entity/UserModel.java]: @gumimaci88 `@OneToMany(mappedBy = 'user')`

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/MergedCalController.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    UserModelRepository userModelRepository;
+
+    @Autowired
+    AuthModelRepository authModelRepository;
+
+    @PostMapping(value = "/calendar")
+    public ResponseEntity postMergedCal(@RequestHeader("X-Client-Token") String clientToken,
+        @RequestBody MergedCalendarFromAndroid mergedCalendarFromAndroid) throws IOException {
+        if (clientToken == null) {
+            return ResponseEntity.badRequest().body("Client token is missing or invalid");
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/MergedCalController.java]: @gumimaci88 401

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/MergedCalController.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        } else {
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/MergedCalController.java]: @gumimaci88 after return you dont need the else

**Comment by malachitedevops** [comment]
Can one of the admins verify this patch?

**Pull request closed without merging**